code/datums/martial/sleeping_carp.dm

```diff
@@ -353,24 +355,25 @@
 	our_held_items += human_carp_user.get_active_held_item()
 	our_held_items += human_carp_user.get_inactive_held_item()
 
-	// We check for wielded objects. If they're not abstract items or exempt items, we
-	for(var/obj/item/possibly_a_held_object as anything in our_held_items)
-		if(possibly_a_held_object.item_flags & (ABSTRACT|HAND_ITEM) && !possibly_a_held_object.block_chance)
-			continue
+	// We check for wielded objects. If they're not abstract items or exempt items, we add their weight as a penalty. And their block chance.
+	if(our_held_items)
```

Comment:
`if(our_held_items)` will always pass, even if it's an empty list. Empty lists are truthy, you want `length(our_held_items)`.
However, the check is redundant anyways. `in null` and `in list()` are both valid - it won't do anything and that's fine.
However however, because you use `as anything` this will still runtime, as the list can have `null` elements (`list(null, null)`), and you are accessing `null.item_flags`.
However however however, why are we even constructing a held items list instead of just using `user.held_items`?

tl;dr
```dm
for(var/obj/item/possibly_a_held_object in human_carp_user.held_items)
```